pkg/route/ingress/ingress.go

```diff
@@ -717,6 +674,64 @@ func generateRouteName(base string) string {
 	return fmt.Sprintf("%s%s", base, utilrand.String(randomLength))
 }
 
+func tlsConfigForIngress(
+	ingress *networkingv1beta1.Ingress,
+	rule *networkingv1beta1.IngressRule,
+	potentiallyNilTLSSecret *corev1.Secret,
+) (_ *routev1.TLSConfig, incomplete bool) {
+	if !needsTLSConfig(ingress, rule) {
+		return nil, false
+	}
+	// Edge: must have cert
```

Comment:
Must this be?  Could we allow edge sans certificate to mean that the ingress uses the ingress controller's default certificate?  (Related: [RFE-339](https://issues.redhat.com/browse/RFE-339) and [BZ1727791](https://bugzilla.redhat.com/show_bug.cgi?id=1727791).)

Reply:
Updated